### ISSUE TITLE
Search does not restore hidden header when using header.autohide

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [squidfunk/mkdocs-material#2233]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the documentation does not mention anything about my problem
  - Only mentions the feature is still 'experimental'
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... there are no open or closed issues that are related to my problem
  - Searched for `autohide`

## Description

When using `header.autohide` , the header is hidden after scrolling down a bit. 

However, if **Search** is invoked by [keyboard shortcut](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#keyboard-shortcuts) keys `[F]` or `[S]`, the header will not reappear and the search bar will be cut off at the top, showing the search results but not the input box.

Sometimes / at some point the header and search box will re-appear.

Please see this gif:

![mr7lYqPaJy](https://user-images.githubusercontent.com/33002073/105563363-19e87980-5d1e-11eb-8b2d-d30deea5f5e0.gif)

### Expected behavior

When invoking search, the auto-hidden header should be displayed again.

### Actual behavior

Header is not always displayed when starting searching.

### Steps to reproduce the bug

1. Use `- header.autohide`
2. Open long page and scroll down until header is auto hidden
3. Hit `S` to trigger search bar and type something, header is not displayed

### Package versions

* Docker Image: `squidfunk/mkdocs-material:6.2.5`

* Python: `Python 3.8.1`
* MkDocs: `mkdocs, version 1.1.2 from /usr/local/lib/python3.8/site-packages/mkdocs (Python 3.8)`
* Material: `Version: 6.2.5`

### Project configuration

Truncated config used when creating the gif shown above:

``` yaml
nav:
  - Übersicht: 
    - Startseite:               index.md
    - Exporter:
      - node_exporter (Linux):  Prometheus/Exporters/node_exporter.md
site_dir: /site
site_name: Monitoring Docs
theme:
  features:
    - header.autohide
    - navigation.tabs
  name: material
```

### System information

* OS: Windows 10 Version 20H2 Build 19042.746
* Browser: Firefox 84.0.2

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting and providing the screen recording! That's indeed unintended, the search should show up.

**Comment by squidfunk** [comment]
Fixed in be1f9c113e2eb91e3f8f2a73aa891aad75113659. I refactored the autohiding logic, which as you discovered is still experimental. This means that there might be other edge cases to consider, so the implementation might still not be complete. Nonetheless, header + search now behave as follows:

https://user-images.githubusercontent.com/932156/105628161-8efda100-5e3b-11eb-93aa-60028289ba90.mp4

It's still not perfect in a way that the page sometimes jumps when focusing the search on the hidden header, but that would need some serious refactoring, as we'd have to delay the focus event until the header is visible. I will consider this for some later refactoring, but for now, I think this should work well enough.
